devito/ir/support/basic.py

```diff
@@ -2,6 +2,7 @@
 from functools import cached_property
 
 from sympy import S
+import sympy
```

Comment:
Nitpick: could you not import `Interval` and `Vector` from SymPy on the line above? It would remove this line and tidy up some of the other code

Reply:
I could, but don't want to, for clarity. We have our own `Interval` and `Vector` too